code/mob/melee_attack_procs.dm

```diff
@@ -93,6 +116,29 @@
 				src.visible_message("<span class='notice'>[src] pats themselves on the back. Feel better, [src].</span>")
 
 	else
+		var/mob/living/M = target
+
+		var/obj/item/implant/projectile/dart/P = locate(/obj/item/implant/projectile/dart) in M.implant
+
+		if (P?.barbed == FALSE)
+			src.visible_message("<span class='alert'><B>[src] pulls a dart out of [target]!</B></span>")
+			P.on_remove(M)
```

Comment:
I was going to suggest that you add their blood DNA to the implant but as it turns out no bullets do that (????)
gonna add that in a bit, might merge conflict sorry
HOWEVER, most bullets leave a forensic profile on the bullet (which will match the gun when scanned), and these should do that too. Looks like you just need to set `forensic_id` on the implant.
Example:
![image](https://user-images.githubusercontent.com/67879783/160267695-e47f07f2-8269-4a24-89d5-27c9a3fccf4d.png)